### DISCUSSION TITLE
Disable/Prevent processing of messages that were sent while the API was inactive

### DISCUSSION THREAD

**Discussion by ZiemlichUndead** [Auties00/WhatsappWeb4j#207] [Q&A] [answered]
Hey, 
looks like new behaviour to me. Since 3.0.5 the API processes all messages that were sent during a downtime. 
I dont want anyone to get spammed at because they wrote my bot when it was offline.
Is there a way to disable this?

**Comment by Auties00** [comment, marked as answer]
That's the correct behavior.
A good approach would be to check the message timestamp against the current time and have a maximum threshold. I'd set it to 10 second for example.

**Reply by ZiemlichUndead** [reply]
Alright. I thought about that but I wondered if there would be a more straight forward solution like toggleing a setting.
Ty ^^

**Reply by Auties00** [reply]
I could overload the onNewMessage listener to have a parameter with an enum type that classifies the message either as history, current message or offline message
